### PR TITLE
DFR-1118: interim hearing collection

### PR DESCRIPTION
**Pull request by rishikrsharma**
### JIRA link (if applicable) ###
DFR-1118

### Change description ###
interim hearing collection

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/InterimHearingContestedAboutToStartHandler.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+        return AboutToStartOrSubmitCallbackResponse.builder().data(caseDetails.getData()).build();
+    }
+
+    private void loadInterimHearing(Map<String, Object> caseData) {
+
+        List<InterimHearingData> interimHearingList = Optional.ofNullable(caseData.get(INTERIM_HEARING_COLLECTION))
+            .map(this::convertToInterimHearingDataList).orElse(new ArrayList<>());
+
+        if (caseData.get(INTERIM_HEARING_TYPE) != null) {
```
Comment (ruban72) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/InterimHearingContestedAboutToStartHandler.java:99]: Maybe refactor to a method here to make the if statement more descriptive e.g isThereAnExistingInterimHearing(caseData)

**Review by ruban72 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/InterimUploadAdditionalDocument.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class InterimUploadAdditionalDocument {
```
Comment (EllisD-B) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/InterimUploadAdditionalDocument.java:15]: Why not just use the already existing CaseDocument POJO?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/InterimHearingItems.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class InterimHearingItems {
```
[the remark this answers is not in the document]
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/InterimHearingItems.java]: InterimHearingItems rename to InterimHearingItem

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/InterimUploadAdditionalDocument.java (hunk 2 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/InterimUploadAdditionalDocument.java:15]: Had an issue. So I went with this approach

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/InterimHearingContestedAboutToStartHandler.java (hunk 1 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/InterimHearingContestedAboutToStartHandler.java:60]: This is moved to InterimHearingHelper class

**Review by EllisD-B — APPROVED** [review]

**Pull request merged**